### ISSUE TITLE
in python run.py --is_finetune True process ,  assert torch.sum(movies_gth!=0, dim=(0,1)) == torch.sum((mask_ys == 6), dim=(0,1)) error

### ISSUE THREAD

**Issue by DuTim** [jokieleung/NTRD#5]
## assert error 
```
raceback (most recent call last):
  File "run.py", line 918, in <module>
    loop.train()
  File "run.py", line 470, in train
    output_metrics_gen = self.val(True)
  File "run.py", line 525, in val
    _, _, _, _, gen_loss, mask_loss, info_db_loss, info_con_loss, selection_loss, _, _ = self.model(context.cuda(), response.cuda(), mask_response.cuda(), concept_mask, dbpedia_mask, seed_sets, movie, concept_vec, db_vec, entity_vector.cuda(), rec,movies_gth.cuda(),movie_nums, test=False)
  File "/home/bigdata10/anaconda3/envs/h10yyb/lib/python3.8/site-packages/torch/nn/modules/module.py", line 727, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/home/bigdata10/yyb/Paper/NTRD-main/model.py", line 600, in forward
    assert torch.sum(movies_gth!=0, dim=(0,1)) == torch.sum((mask_ys == 6), dim=(0,1))
AssertionError
```
I don't find a  way to fix the bug.

**Comment by Veason-silverbullet** [comment]
The same.

**Comment by zcy-cqut** [comment]
Hello, have you solved this problem? I also encountered the same problem and didn't know how to solve it.

**Comment by DuTim** [comment]
> Hello, have you solved this problem? I also encountered the same problem and didn't know how to solve it.

I don't find a way to solve the problem

**Comment by xansar** [comment]
The same.

**Comment by jokieleung** [comment]
Hey guys. I'll check this out ASAP. Sorry for the delay cuase I've been working on my thesis these days.

**Comment by xansar** [comment]
Maybe the bug is at class dataset.__init__() and it could be useful to put the line 'self.prepare_word2vec()' under the condition of 'if 'train' in filename:' . 
If not, when creating the dataset for evaluating,  'self.prepare_word2vec()' will work and recreate the file 'word2index_redial.json' (about 5000 words), which will cover the original one. 
And in this new 'word2index_redial.json', the token __MOVIE__'s index is 5, which doesn't correspond to 6 in the run.py and model.py. 
As a result, the responses in val_dataset use 5 to represent \_\_MOVIE\_\_, but the program gets the number of \_\_MOVIE\_\_ in response by counting the frequency of 6. So the torch.sum(movies_gth!=0, dim=(0,1))  == torch.sum((mask_ys == 6), dim=(0,1)) couldn't be equal.

**Comment by zcy-cqut** [comment]
> Hey guys. I'll check this out ASAP. Sorry for the delay cuase I've been working on my thesis these days.

Thank you. We look forward to your solving all the bugs in the code.

**Comment by dayuyang1999** [comment]
same issue here

**Comment by dayuyang1999** [comment]
> 

nice analysis!

However, I found __movie__ token is 6 here. However the error still exists.
![Screen Shot 2022-04-05 at 5 19 03 PM](https://user-images.githubusercontent.com/60718476/161851026-1acd69e9-6ddc-4590-a1fc-97be9df7fa4c.png)

**Comment by dayuyang1999** [comment]
> to

Hi,

very nice analysis. I think you are probably in the right track.

However, I found in , the movie token is still 6:

![Screen Shot 2022-04-05 at 7 41 20 PM](https://user-images.githubusercontent.com/60718476/161868588-cc68828d-0856-4919-98db-2a38368eeb58.png)

**Comment by jokieleung** [comment]
> Maybe the bug is at class dataset.**init**() and it could be useful to put the line 'self.prepare_word2vec()' under the condition of 'if 'train' in filename:' . If not, when creating the dataset for evaluating, 'self.prepare_word2vec()' will work and recreate the file 'word2index_redial.json' (about 5000 words), which will cover the original one. And in this new 'word2index_redial.json', the token **MOVIE**'s index is 5, which doesn't correspond to 6 in the run.py and model.py. As a result, the responses in val_dataset use 5 to represent __MOVIE__, but the program gets the number of __MOVIE__ in response by counting the frequency of 6. So the torch.sum(movies_gth!=0, dim=(0,1)) == torch.sum((mask_ys == 6), dim=(0,1)) couldn't be equal.

Hey guys! Thanks for the feedback from @xansar, the problem is solved and maybe you should re-run the experiments from scratch.
